File: clients/java/src/main/java/model/Vec2Double.java
package model;

import util.StreamUtil;

public class Vec2Double {
    private double x;
    public double getX() { return x; }
    public void setX(double x) { this.x = x; }
    private double y;
    public double getY() { return y; }
    public void setY(double y) { this.y = y; }
    public Vec2Double() {}
    public Vec2Double(double x, double y) {
        this.x = x;
        this.y = y;
    }
    public static Vec2Double readFrom(java.io.InputStream stream) throws java.io.IOException {
        Vec2Double result = new Vec2Double();
        result.x = StreamUtil.readDouble(stream);
        result.y = StreamUtil.readDouble(stream);
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeDouble(stream, x);
        StreamUtil.writeDouble(stream, y);
    }
}

File: clients/scala/src/main/scala/model/Vec2Double.scala
package model

import util.StreamUtil

case class Vec2Double(x: Double, y: Double) {
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeDouble(stream, x)
        StreamUtil.writeDouble(stream, y)
    }
}
object Vec2Double {
    def readFrom(stream: java.io.InputStream): Vec2Double = Vec2Double(
        StreamUtil.readDouble(stream)
        ,
        StreamUtil.readDouble(stream)
        )
}
